### PR TITLE
#10197: fix Skipping 'Configure Map' step breaks context configuration

### PR DESCRIPTION
**Pull request by mahmoudadel54**
## Description
In this PR: fixing the issue of beak context due to missing map config in case of edit an existing context by skip map configure tab

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
#10197 

**What is the current behavior?**
#10197 

**What is the new behavior?**
 Editing an existing context by skip map configure tab is working without break context 

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/selectors/contextcreator.js [hunk 1]
```diff
@@ -50,7 +50,14 @@ export const prefetchedDataSelector = state => get(state, 'contextcreator.prefet
 export const disableImportSelector = state => creationStepSelector(state) !== "general-settings";
 
 export const generateContextResource = (state) => {
-    const mapConfig = mapSelector(state) ? mapSaveSelector(state) : {};
+    // in some cases: mapSelector(state) includes only eventListeners so check if mapSelector includes center prop
```
Comment (dsuren1) [inline web/client/selectors/contextcreator.js:53]:
```suggestion
    // in some cases, 'mapSelector' doesn't include map configuration, so check and add the map config if missing
```

File: web/client/selectors/contextcreator.js [hunk 2]
```diff
@@ -50,7 +50,14 @@ export const prefetchedDataSelector = state => get(state, 'contextcreator.prefet
...
+    const isMapConfigNotChanged = !mapSelector(state)?.center;
+    let mapConfig = {};
+    if (isMapConfigNotChanged) {
+        mapConfig = mapConfigSelector(state) || {};
+    } else {
+        mapConfig = mapSelector(state) ? mapSaveSelector(state) : {};
+    }
```
Comment (dsuren1) [inline web/client/selectors/contextcreator.js:60]:
```suggestion
    const map = mapSelector(state);
    const mapConfig = map ? map?.center ? mapSaveSelector(state) : mapConfigSelector(state) : {};
```

How about something like this?

**Review by dsuren1 — CHANGES_REQUESTED** [review]
@mahmoudadel54 Fix works. However, can you please look into the suggestions, thanks

File: web/client/selectors/contextcreator.js (hunk 2 above)
Comment (mahmoudadel54) [reply web/client/selectors/contextcreator.js:60]: @dsuren1 
for me it is Ok, but some reviewers prefer the if approach as it eases the debug/review step

Comment (dsuren1) [reply web/client/selectors/contextcreator.js:60]: In that case, mapSelector is called twice which is not needed and `isMapConfigNotChanged` is confusing maybe `isMapConfigPresent`

File: web/client/selectors/contextcreator.js [hunk 3]
```diff
@@ -50,7 +50,15 @@ export const prefetchedDataSelector = state => get(state, 'contextcreator.prefet
 export const disableImportSelector = state => creationStepSelector(state) !== "general-settings";
 
 export const generateContextResource = (state) => {
-    const mapConfig = mapSelector(state) ? mapSaveSelector(state) : {};
+    // in some cases, 'mapSelector' doesn't include map configuration, so check and add the map config if missing
+    const map = mapSelector(state);
+    const isMapConfigPresent = !map?.center;
+    let mapConfig = {};
+    if (isMapConfigPresent) {
```
Comment (dsuren1) [inline web/client/selectors/contextcreator.js:57]:
```suggestion
    let mapConfig = {};
    if (!map?.center) {
```

**Review by dsuren1 — CHANGES_REQUESTED** [review]

**Review by dsuren1 — APPROVED** [review]

**Comment by dsuren1** [comment]
@ElenaGallo Kindly test it in DEV and let us know if it can be backported. Thanks!

**Comment by mahmoudadel54** [comment]
> Test passed, @mahmoudadel54 please backport to 2024.01.xx. Thanks
@ElenaGallo 
Backport is done --- > https://github.com/geosolutions-it/MapStore2/pull/10210

**Pull request merged**